crates/jit/src/mmap_vec.rs

```diff
@@ -73,6 +76,31 @@ impl MmapVec {
         }
     }
 
+    /// Creates a new `MmapVec` which is the `path` specified mmap'd into
+    /// memory.
+    ///
+    /// This function will attempt to open the file located at `path` and will
+    /// then use that file to learn about its size and map the full contents
+    /// into memory. This will return an error if the file doesn't exist or if
+    /// it's too large to be fully mapped into memory.
+    pub fn from_file(path: &Path) -> Result<MmapVec> {
+        let file =
+            File::open(path).with_context(|| format!("failed to open: {}", path.display()))?;
```

Comment:
We'll definitely want to use `OpenOptions::share_mode` (with `FILE_SHARE_READ` probably)  to open the file on Windows to prevent write access to the file so another process can't write to it while we have it mapped.